src/Chart.Doughnut.js

```diff
@@ -92,6 +92,10 @@
 		},
 		addData : function(segment, atIndex, silent){
 			var index = atIndex !== undefined ? atIndex : this.segments.length;
+			if ( typeof(segment.color) === "undefined" ) {
+				segment.color = Chart.defaults.global.segmentColorDefault[index];
```

Comment:
Might be safer to use `index % Chart.defaults.global.segmentColorDefault.length` 
It is unlikely that there would be more than 15 segments, but someone could change the array to a shorter one.